### ISSUE TITLE
[java] ArrayIsStoredDirectly false positive with method call

### ISSUE THREAD

**Issue by kevin-wayne** [pmd/pmd#1474]
**Affects PMD Version:** 
PMD 6.3.0

**Rule:**
[ArrayIsStoredDirectly](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly)

**Description:**
False positive when method contains an array argument that has the same name as a method.

Probably related to #1185.

**Code Sample demonstrating the issue:**

```java
public class TestArrayIsStoredDirectly {
    public double energy(int x) {
        return 0.0;
    }

    // if energy[] is replace by energy2[], then no ArrayIsStoredDirectly warning
    private void f(double[] energy) {
        energy[0] = energy(1);
    }
}
```

Rule Property: allowPrivate=false

**Running PMD through:** *[CLI]*
